dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EntryResource.java

```diff
@@ -92,6 +123,96 @@ public Entry updateAliases(@ApiParam(hidden = true) @Auth User user,
         return this.toolDAO.findCollectionsByEntryId(entry.getId());
     }
 
+    @PUT
+    @Path("/{id}/topic")
+    @Timed
+    @RolesAllowed({ "curator" })
+    @UnitOfWork
+    @ApiOperation(value = "Create a discourse topic for an entry.", authorizations = {
+            @Authorization(value = JWT_SECURITY_DEFINITION_NAME) }, response = Entry.class)
+    public Entry setDiscourseTopic(@ApiParam(hidden = true) @Auth User user,
+            @ApiParam(value = "The id of the entry to add a topic to.", required = true) @PathParam("id") Long id,
+            @ApiParam(value = "The id of the category to add a topic to, defaults to Automatic Tool and Workflow Threads(6).", defaultValue = "6") @QueryParam("categoryId") Integer categoryId,
```

Comment:
I would have a space before `(6)`. 

What other ids are there and how would you discover them? May be beyond the scope of this PR and Swagger annotations, but doesn't seem that intuitive on how to use.

Reply:
The other IDs are available on Discourse, but this endpoint should only work with 6 or 9, which are the Auto Tools and Workflows category and the Testing Category. I'll update this to use allowable values.